examples/passport/logged-in-user-with-nextjs/src/app/link-external-wallet/page.tsx

```
@@ -0,0 +1,286 @@
+              type: "address"
+            },
+            {
+              name: "condition",
+              type: "string"
+            },
+            {
+              name: "nonce",
+              type: "string"
+            }
+          ]
+        },
+        primaryType: "LinkWallet",
+        domain: {
+          chainId: 1, // Must be set to 1 for Ethereum Mainnet
+        },
+        message: {
+          walletAddress: metamaskAddress,
+          immutablePassportAddress: passportAddress,
+          condition: "I agree to link this wallet to my Immutable Passport account.",
+          nonce
+        }
+      }
+
+      if (typeof window === 'undefined' || !window.ethereum) {
+        throw new Error('MetaMask not installed');
+      }
+
+      let signature: string;
+
+      try {
+         // Metamask must be connected to Ethereum Mainnet
+        await window.ethereum.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: '0x1' }], // chainId must be in hexadecimal format
+        });
+
+        // Now request the signature
+        signature = await window.ethereum.request({
+          method: 'eth_signTypedData_v4',
+          params: [metamaskAddress, JSON.stringify(dataToSign)]
+        });
+
+      } catch (error) {
+        console.error('Error signing message:', error);
+        setLinkingError('Failed to sign message');
+        setLinkingStatus('Make sure you have MetaMask installed and connected to Ethereum Mainnet');
+        setIsLinking(false);
+        return
+      }
+     
+      setLinkingStatus('Linking wallet...');
+      
+      // Call the linkExternalWallet method to link the wallet
+      const result = await passportInstance.linkExternalWallet({
+        type: "External",
+        walletAddress: metamaskAddress,
+        signature,
+        nonce
+      });
+      
+      const linkedAddresses = await passportInstance.getLinkedAddresses();
+      setLinkedAddresses(linkedAddresses);
```

If a user already had a linked external wallet and tries to link it again, the linkedAddresses state will be null even though it's not actually null. Do we want to make it not null to show the actual values? 

- This is because an error will be thrown in line 179 and 187-189 wont run

---

In that case, should we check first if the wallet address is already linked before trying to link it again?

---

> In that case, should we check first if the wallet address is already linked before trying to link it again?

Yep, that makes sense. Should we throw an error if a user had a linked external wallet prior to the "linkExternalWallet" method call?

---

This depends on what behavior we think is the right one to surface in our example, do we want developers implementing the passport functionality to show an error, or should it just continue on and ignore linking it if it's already linked? Either way is appropriate but depends on what the dev rel team thinks is most optimal 
cc @proletesseract